plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FlexFacetInstallDelegate.java

```diff
@@ -16,6 +16,46 @@
 
 package com.google.cloud.tools.eclipse.appengine.facets;
 
+import com.google.cloud.tools.eclipse.appengine.deploy.flex.FlexDeployPreferences;
+import com.google.cloud.tools.eclipse.util.io.ResourceUtils;
+import com.google.cloud.tools.eclipse.util.templates.appengine.AppEngineTemplateUtility;
+import java.io.ByteArrayInputStream;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+
 public class FlexFacetInstallDelegate extends AppEngineFacetInstallDelegate {
+  @Override
+  public void execute(IProject project,
+                      IProjectFacetVersion version,
+                      Object config,
+                      IProgressMonitor monitor) throws CoreException {
+    super.execute(project, version, config, monitor);
+    createConfigFiles(project, monitor);
+  }
+
+  // TODO: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1640
+  // TODO: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1642
+  private void createConfigFiles(IProject project, IProgressMonitor monitor) throws CoreException {
+    SubMonitor progress = SubMonitor.convert(monitor, 10);
```

Comment:
Nit: seems like the monitor value should be 4, instead of 10. Also there is a short-circuit branch in the middle of this method, so maybe it needs to be moved after that branch or advance the whole tick before short-circuiting?